packages/@react-aria/utils/src/scrollIntoView.ts

```diff
@@ -63,3 +75,37 @@ function relativeOffset(ancestor: HTMLElement, child: HTMLElement, axis: 'left'|
   }
   return sum;
 }
+
+/**
+ * Scrolls the `targetElement` so it is visible in the viewport. Accepts an optional `opts.containingElement`
+ * that will be centered in the viewport prior to scrolling the targetElement into view. If scrolling is prevented on
+ * the body (e.g. targetElement is in a popover), this will only scroll the scroll parents of the targetElement up to but not including the body itself.
+ */
+export function scrollIntoViewport(targetElement: Element, opts?: ScrollIntoViewportOpts) {
+  if (document.contains(targetElement)) {
+    let root = document.scrollingElement || document.documentElement;
+    let isScrollPrevented = window.getComputedStyle(root).overflow === 'hidden';
+    // If scrolling is not currently prevented then we aren’t in a overlay nor is a overlay open, just use element.scrollIntoView to bring the element into view
+    if (!isScrollPrevented) {
+      let {left: originalLeft, top: originalTop} = targetElement.getBoundingClientRect();
+
+      // use scrollIntoView({block: 'nearest'}) instead of .focus to check if the element is fully in view or not since .focus()
+      // won't cause a scroll if the element is already focused and doesn't behave consistently when an element is partially out of view horizontally vs vertically
+      targetElement?.scrollIntoView?.({block: 'nearest'});
+      let {left: newLeft, top: newTop} = targetElement.getBoundingClientRect();
+      // Account for sub pixel differences from rounding
+      if ((Math.abs(originalLeft - newLeft) > 1) || (Math.abs(originalTop - newTop) > 1)) {
+        opts?.containingElement?.scrollIntoView?.({block: 'center', inline: 'center'});
+        targetElement.scrollIntoView?.({block: 'nearest'});
+      }
+    } else {
+      let scrollParent = getScrollParent(targetElement);
+      // If scrolling is prevented, we don't want to scroll the body since it might move the overlay partially offscreen and the user can't scroll it back into view.
+      while (targetElement && scrollParent && targetElement !== root && scrollParent !== root) {
```

Comment:
Why is this a `while` and not an `if`?

Reply:
This is a while because we iterate through every scroll body and adjust the scroll positioning of the target in such a way that it is "visible" in its scrollable parent. The scenario this covers is if you had a bunch of nested scroll regions and each were scrolled in such a way that they obscured the target and thus just scrolling the original target and its parent scroll region wouldn't be sufficient.